block/gs/compose.go

```diff
@@ -0,0 +1,33 @@
+package gs
+
+import (
+	"fmt"
+)
+
+const MaxPartsInCompose = 32
+
+type ComposeFunc func(target string, parts []string) error
+
+func ComposeAll(target string, parts []string, composeFunc ComposeFunc) error {
+	for layer := 1; len(parts) > MaxPartsInCompose; layer++ {
+		var nextParts []string
+		for i := 0; i < len(parts); i += MaxPartsInCompose {
+			chunkSize := len(parts) - i
+			if chunkSize > MaxPartsInCompose {
+				chunkSize = MaxPartsInCompose
+			}
+			chunk := parts[i : i+chunkSize]
+			if chunkSize == 1 || (chunkSize < MaxPartsInCompose && len(nextParts)+chunkSize < MaxPartsInCompose) {
+				nextParts = append(nextParts, chunk...)
```

Comment:
can be "len(nextParts)+chunkSize =< MaxPartsInCompose". or remove the first condition

Reply:
In case `len(nextParts)` is 0 and `chunkSize` is 32 - the condition will be true and we don't want to append in this case.
I can remove the chunkSize == 1 check - I just wanted to be sure that there is no option we compose a chunk of one.

Reply:
Got it - fixed